{
	"id": "CVE-2021-37651",
	"vulnerabilities": [
		{
			"content": {
				"id": "CVE-2021-37651",
				"description": "TensorFlow is an end-to-end open source platform for machine learning. In affected versions the implementation for `tf.raw_ops.FractionalAvgPoolGrad` can be tricked into accessing data outside of bounds of heap allocated buffers. The [implementation](https://github.com/tensorflow/tensorflow/blob/f24faa153ad31a4b51578f8181d3aaab77a1ddeb/tensorflow/core/kernels/fractional_avg_pool_op.cc#L205) does not validate that the input tensor is non-empty. Thus, code constructs an empty `EigenDoubleMatrixMap` and then accesses this buffer with indices that are outside of the empty area. We have patched the issue in GitHub commit 0f931751fb20f565c4e94aa6df58d54a003cdb30. The fix will be included in TensorFlow 2.6.0. We will also cherrypick this commit on TensorFlow 2.5.1, TensorFlow 2.4.3, and TensorFlow 2.3.4, as these are also affected and still in supported range.",
				"severity": [
					{
						"type": "cvss_v2",
						"source": "nvd@nist.gov",
						"cvss_v2": {
							"vector": "AV:L/AC:L/Au:N/C:P/I:P/A:P",
							"base_score": 4.6,
							"nvd_base_severity": "MEDIUM",
							"temporal_score": 4.6,
							"nvd_temporal_severity": "MEDIUM",
							"environmental_score": 4.6,
							"nvd_environmental_severity": "MEDIUM"
						}
					},
					{
						"type": "cvss_v31",
						"source": "nvd@nist.gov",
						"cvss_v31": {
							"vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H",
							"base_score": 7.8,
							"base_severity": "HIGH",
							"temporal_score": 7.8,
							"temporal_severity": "HIGH",
							"environmental_score": 7.8,
							"environmental_severity": "HIGH"
						}
					},
					{
						"type": "cvss_v31",
						"source": "security-advisories@github.com",
						"cvss_v31": {
							"vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:N",
							"base_score": 7.1,
							"base_severity": "HIGH",
							"temporal_score": 7.1,
							"temporal_severity": "HIGH",
							"environmental_score": 7.1,
							"environmental_severity": "HIGH"
						}
					}
				],
				"cwe": [
					{
						"source": "nvd@nist.gov",
						"cwe": [
							"CWE-787"
						]
					},
					{
						"source": "security-advisories@github.com",
						"cwe": [
							"CWE-125"
						]
					}
				],
				"references": [
					{
						"source": "af854a3a-2127-422b-91ae-364da2661108",
						"url": "https://github.com/tensorflow/tensorflow/commit/0f931751fb20f565c4e94aa6df58d54a003cdb30"
					},
					{
						"source": "af854a3a-2127-422b-91ae-364da2661108",
						"url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-hpv4-7p9c-mvfr"
					},
					{
						"source": "nvd.nist.gov",
						"url": "https://nvd.nist.gov/vuln/detail/CVE-2021-37651"
					},
					{
						"source": "security-advisories@github.com",
						"url": "https://github.com/tensorflow/tensorflow/commit/0f931751fb20f565c4e94aa6df58d54a003cdb30"
					},
					{
						"source": "security-advisories@github.com",
						"url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-hpv4-7p9c-mvfr"
					}
				],
				"published": "2021-08-12T21:15:08.17Z",
				"modified": "2024-11-21T06:15:36.86Z"
			},
			"segments": [
				{
					"ecosystem": "cpe"
				}
			]
		}
	],
	"detections": [
		{
			"ecosystem": "cpe",
			"conditions": [
				{
					"criteria": {
						"operator": "OR",
						"criterias": [
							{
								"operator": "OR",
								"criterias": [
									{
										"operator": "OR",
										"criterias": [
											{
												"operator": "OR",
												"criterions": [
													{
														"type": "version",
														"version": {
															"vulnerable": true,
															"fix_status": {
																"class": "unknown"
															},
															"package": {
																"type": "cpe",
																"cpe": "cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*"
															},
															"affected": {
																"type": "semver",
																"range": [
																	{
																		"lt": "2.3.4",
																		"ge": "2.3.0"
																	}
																]
															}
														}
													}
												]
											},
											{
												"operator": "OR",
												"criterions": [
													{
														"type": "version",
														"version": {
															"vulnerable": true,
															"fix_status": {
																"class": "unknown"
															},
															"package": {
																"type": "cpe",
																"cpe": "cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*"
															},
															"affected": {
																"type": "semver",
																"range": [
																	{
																		"lt": "2.4.3",
																		"ge": "2.4.0"
																	}
																]
															}
														}
													}
												]
											},
											{
												"operator": "OR",
												"criterions": [
													{
														"type": "version",
														"version": {
															"vulnerable": true,
															"fix_status": {
																"class": "unknown"
															},
															"package": {
																"type": "cpe",
																"cpe": "cpe:2.3:a:google:tensorflow:2.5.0:*:*:*:*:*:*:*"
															}
														}
													}
												]
											},
											{
												"operator": "OR",
												"criterions": [
													{
														"type": "version",
														"version": {
															"vulnerable": true,
															"fix_status": {
																"class": "unknown"
															},
															"package": {
																"type": "cpe",
																"cpe": "cpe:2.3:a:google:tensorflow:2.6.0:rc0:*:*:*:*:*:*"
															}
														}
													}
												]
											},
											{
												"operator": "OR",
												"criterions": [
													{
														"type": "version",
														"version": {
															"vulnerable": true,
															"fix_status": {
																"class": "unknown"
															},
															"package": {
																"type": "cpe",
																"cpe": "cpe:2.3:a:google:tensorflow:2.6.0:rc1:*:*:*:*:*:*"
															}
														}
													}
												]
											},
											{
												"operator": "OR",
												"criterions": [
													{
														"type": "version",
														"version": {
															"vulnerable": true,
															"fix_status": {
																"class": "unknown"
															},
															"package": {
																"type": "cpe",
																"cpe": "cpe:2.3:a:google:tensorflow:2.6.0:rc2:*:*:*:*:*:*"
															}
														}
													}
												]
											}
										]
									}
								]
							}
						]
					}
				}
			]
		}
	],
	"data_source": {
		"id": "nvd-api-cve",
		"raws": [
			"vuls-data-raw-nvd-api-cve/2021/CVE-2021-37651.json"
		]
	}
}
